apiary-metastore-events/kafka-metastore-events/kafka-events-common/src/main/java/com/expediagroup/apiary/extensions/events/metastore/common/io/jackson/SkewedInfoDeserializer.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (C) 2018-2019 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.apiary.extensions.events.metastore.common.io.jackson;
+
+import static com.fasterxml.jackson.core.JsonToken.END_OBJECT;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.hadoop.hive.metastore.api.SkewedInfo;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+
+public class SkewedInfoDeserializer extends StdDeserializer<SkewedInfo> {
```

Comment:
Can someone remind me what this thing is and why we need it?

Reply:
IIRC, I think `SkewedInfo` is a thrift type which is being used by one of the fields being serialized within the Hive Event. So, we needed this custom deserializer to handle it. I have tried removing it earlier but deserialization fails then.

Reply:
Will leave then as it's needed

Reply:
OK, makes sense.